core/src/main/scala/akka/persistence/dynamodb/internal/QueryDao.scala

```diff
@@ -206,16 +210,34 @@ import software.amazon.awssdk.services.dynamodb.model.QueryRequest
 
       val publisher = client.queryPaginator(req)
 
+      def getTimestamp(item: JMap[String, AttributeValue]): Instant =
+        InstantFactory.fromEpochMicros(item.get(Timestamp).n().toLong)
+
+      val logName = s"itemsBySlice - $entityTypeSlice - ${if (backtracking) "backtracking" else "query"}"
+
+      def logQueryResponse: QueryResponse => String = response => {
+        if (response.hasItems && !response.items.isEmpty) {
+          val items = response.items
+          val count = items.size
+          val first = getTimestamp(items.get(0))
+          val last = getTimestamp(items.get(items.size - 1))
+          val scanned = response.scannedCount
+          val hasMore = response.hasLastEvaluatedKey && !response.lastEvaluatedKey.isEmpty
+          s"query response page with [$count] events between [$first - $last] (scanned [$scanned], has more [$hasMore])"
+        } else "empty query response page"
+      }
+
       Source
         .fromPublisher(publisher)
+        .log(logName, logQueryResponse)(logging)
```

Comment:
Could we propagate the logPrefix to the dao so that we try to have the same format, or at least similar prefix format here?
```
s"[$entityType] eventsBySlice [$slice]: "
```

Is there some added value of using stream `log` or shall we just add ordinary logging inside the mapConcat?
For example, stream fail will be logged at error level if we don't adjust the levels.

Reply:
I'll update the log prefix to be the same pattern. Yes, using the stream logging intentional here, so we can see the completions in particular. We could adjust failures to be debug as well, if we think error would be a problem.